src/FrequencyMapAgent.cpp

```diff
@@ -364,22 +364,33 @@ namespace geopm
     std::vector<std::pair<std::string, std::string> > FrequencyMapAgent::report_host(void) const
     {
         std::vector<std::pair<std::string, std::string> > result;
-        std::ostringstream oss;
-        oss << std::setprecision(M_PRECISION) << std::scientific;
-        for (const auto &region : m_hash_freq_map) {
-            oss << "\n    0x" << std::hex << std::setfill('0') << std::setw(16) << std::fixed;
-            oss << region.first;
-            oss << std::setfill('\0') << std::setw(0) << std::scientific;
-            oss << ": " << region.second;
-        }
-        for (const auto &region : m_default_freq_hash) {
-            oss << "\n    0x" << std::hex << std::setfill('0') << std::setw(16) << std::fixed;
-            oss << region;
-            oss << std::setfill('\0') << std::setw(0) << std::scientific;
-            oss << ": " << m_default_freq;
+
+        std::map<uint64_t, double> full_map(m_hash_freq_map);
+        std::for_each(m_default_freq_hash.begin(), m_default_freq_hash.end(), [&](uint64_t key)
+        {
+            full_map.insert({ key, m_default_freq });
+        });
+
+        std::map<std::string, Json> temp_map;
+        for (const auto &region : full_map)
+        {
+            // Json object(int(region.first));
+            // std::string key_string = object.dump();
+            // key_string.erase(std::remove(key_string.begin(), key_string.end(), '"'), key_string.end());
+            // temp_map[key_string] = region.second;
+            std::ostringstream key_ss;
+            key_ss << "0x" << std::hex << std::setfill('0') << std::setw(16) << std::fixed;
+            key_ss << region.first;
+            std::ostringstream value_ss;
+            value_ss << std::setfill('\0') << std::setw(0) << std::scientific;
+            value_ss << region.second;
+            std::string key_string = key_ss.str();
+            key_string.erase(std::remove(key_string.begin(), key_string.end(), '"'), key_string.end());
+            temp_map[key_string] = value_ss.str();
         }
-        oss << "\n";
-        result.push_back(std::make_pair("Frequency map", oss.str()));
+        Json my_json = Json(temp_map);
```

Comment:
I'd prefer to see the original method of using a single `std::ostringstream` used here instead of splitting up the region and the frequency strings into separate streams.

You should be able to avoid that `erase` call, and it's overall less code.  The way you have it here will also result in improper indentation in the report which will cause a parsing problem in the `RawReport` Python code.

The resulting output string from the `std::ostringstream` could still be parsed by json11 if that is desired.